Address Safer CPP warnings in MediaStream

eric-carlson opened the pull request:
#### 8368791c63c9b7951c406ce9f820b11d99ff5d25
<pre>
Address Safer CPP warnings in MediaStream
<a href="https://bugs.webkit.org/show_bug.cgi?id=289720">https://bugs.webkit.org/show_bug.cgi?id=289720</a>
<a href="https://rdar.apple.com/146969452">rdar://146969452</a>

Reviewed by Chris Dumez.

* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::clone):
(WebCore::MediaStream::didAddTrack):
(WebCore::MediaStream::startProducingData):
* Source/WebCore/Modules/mediastream/MediaStream.h:

* Source/WebCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:

* Source/WebCore/platform/mediastream/MediaStreamPrivate.cpp:
(WebCore::MediaStreamPrivate::MediaStreamPrivate):
(WebCore::MediaStreamPrivate::~MediaStreamPrivate):
(WebCore::MediaStreamPrivate::forEachTrack const):
(WebCore::MediaStreamPrivate::forEachTrack):
(WebCore::MediaStreamPrivate::addTrack):
(WebCore::MediaStreamPrivate::startProducingData):
(WebCore::MediaStreamPrivate::stopProducingData):
(WebCore::MediaStreamPrivate::isProducingData const):
(WebCore::MediaStreamPrivate::hasVideo const):
(WebCore::MediaStreamPrivate::hasAudio const):
(WebCore::MediaStreamPrivate::muted const):
(WebCore::MediaStreamPrivate::intrinsicSize const):
(WebCore::MediaStreamPrivate::updateActiveVideoTrack):
(WebCore::MediaStreamPrivate::monitorOrientation):
* Source/WebCore/platform/mediastream/MediaStreamPrivate.h:

Canonical link: <a href="https://commits.webkit.org/293155@main">https://commits.webkit.org/293155@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c1609b5116900cfe0634cd6c2638173a3dba7311

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97935 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17560 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7781 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103051 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48465 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/99980 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17854 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26013 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74573 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31756 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100938 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13528 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88503 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54930 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/replace-get-all-cookies (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13305 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6435 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47907 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83281 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6508 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105429 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25015 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18239 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83572 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25387 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84684 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83016 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20987 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27647 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5326 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18639 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24976 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30145 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24798 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28112 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26372 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/mediastream/MediaStream.cpp:
```diff
@@ -123,13 +123,17 @@ RefPtr<MediaStream> MediaStream::clone()
 {
     ALWAYS_LOG(LOGIDENTIFIER);
 
+    RefPtr document = protectedDocument();
```
Per the style guide, we only use `protectedFoo()` functions to avoid local variables, not to initialize local variables. There is no perf / safety benefit to be calling the longer function name here.

```
RefPtr document = this->document();
```
is what we would usually use.

cdumez commented on Source/WebCore/Modules/mediastream/MediaStream.cpp:
```diff
@@ -277,7 +281,7 @@ void MediaStream::mediaCanStart(Document& document)
 void MediaStream::startProducingData()
 {
-    Document* document = this->document();
+    RefPtr document = protectedDocument();
```
Ditto.

cdumez commented on Source/WebCore/Modules/mediastream/MediaStream.cpp:
```diff
@@ -374,6 +378,11 @@ Document* MediaStream::document() const
     return downcast<Document>(scriptExecutionContext());
 }
 
+RefPtr<Document> MediaStream::protectedDocument() const
```
I doubt we need this. We rarely need `protectedFoo()` functions for things that return a RefPtr (and not a Ref). This is because they usually need a null check and thus a local variable.

cdumez reviewed (approved):
LGTM with nits.

eric-carlson committed:
Address Safer CPP warnings in MediaStream
https://bugs.webkit.org/show_bug.cgi?id=289720
rdar://146969452

Reviewed by Chris Dumez.

* Source/WebCore/Modules/mediastream/MediaStream.cpp:
(WebCore::MediaStream::clone):
(WebCore::MediaStream::didAddTrack):
(WebCore::MediaStream::startProducingData):
* Source/WebCore/Modules/mediastream/MediaStream.h:

* Source/WebCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:

* Source/WebCore/platform/mediastream/MediaStreamPrivate.cpp:
(WebCore::MediaStreamPrivate::MediaStreamPrivate):
(WebCore::MediaStreamPrivate::~MediaStreamPrivate):
(WebCore::MediaStreamPrivate::forEachTrack const):
(WebCore::MediaStreamPrivate::forEachTrack):
(WebCore::MediaStreamPrivate::addTrack):
(WebCore::MediaStreamPrivate::startProducingData):
(WebCore::MediaStreamPrivate::stopProducingData):
(WebCore::MediaStreamPrivate::isProducingData const):
(WebCore::MediaStreamPrivate::hasVideo const):
(WebCore::MediaStreamPrivate::hasAudio const):
(WebCore::MediaStreamPrivate::muted const):
(WebCore::MediaStreamPrivate::intrinsicSize const):
(WebCore::MediaStreamPrivate::updateActiveVideoTrack):
(WebCore::MediaStreamPrivate::monitorOrientation):
* Source/WebCore/platform/mediastream/MediaStreamPrivate.h:

Canonical link: https://commits.webkit.org/293155@main
